projects/fab-speed-dial/src/lib/fab-speed-dial.spec.ts

```diff
@@ -103,12 +103,60 @@ describe('FabSpeedDial', () => {
         expect(fixture.componentInstance.fabActions.show).toHaveBeenCalled();
     });
 
+    it('should click on document testElement to hide all fabActions', () => {
+        const fixture = TestBed.createComponent(TestAppComponent);
+        const testComponent = fixture.debugElement.componentInstance;
+
+        const actionsSpy = spyOn(fixture.componentInstance.fabSpeedDial, 'setActionsVisibility').and.callThrough();
+        spyOn(fixture.componentInstance.fabActions, 'show').and.callThrough();
+        spyOn(fixture.componentInstance.fabActions, 'hide').and.callThrough();
+
+        testComponent.open = true;
+        testComponent.closeOnOutsideClick = true;
+        fixture.detectChanges();
+
+        expect(fixture.componentInstance.fabSpeedDial.setActionsVisibility).toHaveBeenCalled();
+        expect(fixture.componentInstance.fabActions.show).toHaveBeenCalled();
+        actionsSpy.calls.reset();
+
+        const actionButton = fixture.debugElement.query(By.css('.testElement'));
+        actionButton.nativeElement.click();
+        fixture.detectChanges();
+
+        expect(fixture.componentInstance.fabSpeedDial.setActionsVisibility).toHaveBeenCalled();
+        expect(fixture.componentInstance.fabActions.hide).toHaveBeenCalled();
+    });
+
+    it('should click on document testElement but not hide fabActions', () => {
+        const fixture = TestBed.createComponent(TestAppComponent);
+        const testComponent = fixture.debugElement.componentInstance;
+
+        const actionsSpy = spyOn(fixture.componentInstance.fabSpeedDial, 'setActionsVisibility').and.callThrough();
+        spyOn(fixture.componentInstance.fabActions, 'show').and.callThrough();
+        spyOn(fixture.componentInstance.fabActions, 'hide').and.callThrough();
+
+        testComponent.open = true;
+        testComponent.closeOnOutsideClick = false;
+        fixture.detectChanges();
+
+        expect(fixture.componentInstance.fabSpeedDial.setActionsVisibility).toHaveBeenCalled();
+        expect(fixture.componentInstance.fabActions.show).toHaveBeenCalled();
+        actionsSpy.calls.reset();
+
+        const actionButton = fixture.debugElement.query(By.css('.testElement'));
+        actionButton.nativeElement.click();
+        fixture.detectChanges();
+
+        expect(fixture.componentInstance.fabSpeedDial.setActionsVisibility).not.toHaveBeenCalled();
+        expect(fixture.componentInstance.fabActions.hide).not.toHaveBeenCalled();
+    });
+
 });
 
 /** Test component that contains an fab speed dial buttons */
 @Component({
-    template: `
-        <eco-fab-speed-dial [direction]="direction" [(open)]="open" #fabSpeedDial>
+    template: `<div>
+        <eco-fab-speed-dial [direction]="direction" [(open)]="open" [closeOnOutsideClick]="closeOnOutsideClick" #fabSpeedDial>
```

Comment:
Without configuration, this attribute would disappear too

```suggestion
        <eco-fab-speed-dial [direction]="direction" [(open)]="open" #fabSpeedDial>
```